packages/clarity-js/src/interaction/pointer.ts

```diff
@@ -58,13 +60,26 @@ function touch(event: Event, root: Node, evt: TouchEvent): void {
             x = x && frame ? x + Math.round(frame.offsetLeft) : x;
             y = y && frame ? y + Math.round(frame.offsetTop) : y;
 
-            // identifier is 0-based, unique for each touch point and resets when all fingers are lifted
-            // that is not a part of the spec, but it is how it is implemented in browsers
-            // tested in Chromium-based browsers as well as Firefox
+            // We cannot rely on identifier to determine primary touch as its value doesn't always start with 0.
+            // Safari/Webkit uses the address of the UITouch object as the identifier value for each touch point.
             const id = "identifier" in entry ? entry["identifier"] : undefined;
 
+            switch(event) {
+                case Event.TouchStart:
+                    if (activeTouchPointIds.size === 0) {
+                        activeTouchPointId = id;
+                    }
+                    activeTouchPointIds.add(id);
+                    break;
+                case Event.TouchEnd:
+                case Event.TouchCancel:
+                    activeTouchPointIds.delete(id);
+                    break;
+            }
+            const isPrimary: boolean = activeTouchPointId === id;
+
             // Check for null values before processing this event
-            if (x !== null && y !== null) { handler({ time: t, event, data: { target: target(evt), x, y, id } }); }
+            if (x !== null && y !== null) { handler({ time: t, event, data: { target: target(evt), x, y, id, isPrimary } }); }
```

Comment:
don't you need to add something into encode so this value actually gets sent?

Reply:
yes! added them. thanks!